client/rpc/src/lib.rs

```diff
@@ -183,6 +213,60 @@ where
         Ok(file_metadata)
     }
 
+    async fn save_file_to_disk(
+        &self,
+        file_key: H256,
+        file_path: String,
+    ) -> RpcResult<SaveFileToDisk> {
+        // Acquire FileStorage read lock.
+        let read_file_storage = self.file_storage.read().await;
+
+        let file_key_hash: HasherOutT<T> = TryFrom::<[u8; 32]>::try_from(file_key.to_fixed_bytes())
+            .map_err(|_| into_rpc_error("Invalid file key hash"))?;
+
+        // Retrieve file metadata from File Storage.
+        let file_metadata = match read_file_storage.get_metadata(&file_key_hash) {
+            Ok(metadata) => metadata,
+            Err(FileStorageError::FileDoesNotExist) => return Ok(SaveFileToDisk::FileNotFound),
```

Comment:
Shouldn't we return `Err` here as well?

Reply:
Why? The result is an enum which defines 3 states: we don't have the file, we have it partially (and we return a count of total and current chunk count) and we have it all (in which case we save it). This case is when we do not have the file.

Reply:
It is not really an error to not have a requested file and some consumer of this RPC might want for example to monitor the progress of the file being download.

Reply:
> some consumer of this RPC might want for example to monitor the progress of the file being download.

makes sense!